src/Microsoft.Sbom.Api/Executors/ILicenseInformationService2.cs

```diff
@@ -0,0 +1,12 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Microsoft.Sbom.Api.Executors;
+
+public interface ILicenseInformationService2 : ILicenseInformationService
```

Comment:
I don't love the name ILicenseInformationService2  here- can we have something more descriptive? Maybe ILicenseInformationTimeoutableService?

Reply:
Alternatively- we could just replace the old ILicenseInformationService and bump the API version by a minor version instead of our usual patch increase. I would prefer that approach but open to opinions @DaveTryon

Reply:
>I don't love the name ILicenseInformationService2 here- can we have something more descriptive? Maybe ILicenseInformationTimeoutableService?

This is a well-established pattern (in the Windows world, at least). You're right that the description is lacking, but trying to be more descriptive leads to weirdness in later iterations. You end up with ILicenseInformationServiceWithX, then ILicenseInformationServiceWithXandY, then ILicenseInformationServiceWithXandYandZ.

> Alternatively- we could just replace the old ILicenseInformationService and bump the API version by a minor version instead of our usual patch increase.

If someone has their own implementation of `ILicenseInformationService`, then changing the interface in any way would be a breaking change to them. As such, we would need to bump the major version. Extending the interface gives us the functionality without breaking any existing customers. Since this is new functionality, bumping the minor version is probably appropriate.

Reply:
Gotcha, that's fine with me